src/directives/compose_area.ts

```diff
@@ -121,26 +126,63 @@ export default [
                                 $log.warn(logTag, 'Unhandled node:', node);
                         }
                     }
-                    const textMessageData: threema.TextMessageData = {text: text.trim()};
-                    // send as array
-                    return scope.submit('text', [textMessageData]);
+                    return new Promise((resolve, reject) => {
+                        let submitTexts = (strings: string[]) => {
+                            let messages: threema.TextMessageData[] = [];
+                            strings.forEach((piece: string) => {
+                                messages.push({
+                                    text: piece,
+                                } as threema.TextMessageData);
+                            });
+                            scope.submit('text', messages)
+                                .then(() => {
+                                    resolve();
+                                })
+                                .catch((error) => {
+                                    reject(error);
+                                });
+                        };
+
+                        let fullText = text.trim();
+                        if (fullText.length > scope.maxTextLength) {
+                            let pieces: string[] = stringService.byteChunk(fullText, scope.maxTextLength, 20);
```

Comment:
Maybe 50 would be better as default offset, to prevent splitting long words? (The default could also be changed directly in the string service...)